document/src/main/java/com/yahoo/document/json/LazyTokenBuffer.java

```diff
@@ -0,0 +1,61 @@
+package com.yahoo.document.json;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+
+import java.io.IOException;
+import java.util.function.Supplier;
+
+/**
+ * A {@link TokenBuffer} which only buffers tokens when needed, i.e., when peeking.
+ *
+ * @author jonmv
+ */
+public class LazyTokenBuffer extends TokenBuffer {
```

Comment:
Could the roles be reversed? Yes, certainly—the old `TokenBuffer` could indeed be replicated by simply peeking at a lazy one until it produced no more tokens. But there was so much unrelated wiring to change, so in the end, I opted for just getting the job done. I look forward to removing lots of this code, immediately when we're allowed to remove the old feeder support.

Reply:
I think ideally we'd remove the non-lazy token buffer entirely. This would also remove the need for the current  intimate (and not necessarily obvious) member variable mutations into the super-type.

Reply:
Yes ... and that also requires a silly `consume()` or similar, called on the token buffer prior to reading any _other_ field from the JSON parser, after handing the parser to said buffer; it's doable, but I considered it to be even more confusing than this, where the different buffers at least have completely separate uses.